frontend/src/container/MySettings/TimezoneAdaptation/TimezoneAdaptation.tsx

```diff
@@ -0,0 +1,78 @@
+import './TimezoneAdaptation.styles.scss';
+
+import { Color } from '@signozhq/design-tokens';
+import { Switch } from 'antd';
+import { Delete } from 'lucide-react';
+import { useTimezone } from 'providers/Timezone';
+import { useMemo, useState } from 'react';
+
+function TimezoneAdaptation(): JSX.Element {
+	const { timezone, browserTimezone, updateTimezone } = useTimezone();
+
+	const isTimezoneOverridden = useMemo(
+		() => timezone?.offset !== browserTimezone.offset,
+		[timezone, browserTimezone],
+	);
+
+	const [isAdaptationEnabled, setIsAdaptationEnabled] = useState(true);
+
+	const getSwitchStyles = (): React.CSSProperties => ({
+		backgroundColor:
+			isAdaptationEnabled && isTimezoneOverridden ? Color.BG_AMBER_400 : undefined,
+	});
+
+	const handleOverrideClear = (): void => {
+		updateTimezone(browserTimezone);
+	};
+
+	return (
+		<div className="timezone-adaption">
+			<div className="timezone-adaption__header">
+				<h2 className="timezone-adaption__title">Adapt to my timezone</h2>
```

Comment:
Not sure about the wording of this. @GeekBoySupreme : Can we use simple words here?

Current TimeZone
We can also provide an option to update the timezone from here ?

Note: Though the data is stored in UTC, currently we show data wrt local time zone.